### PR TITLE
Adds Parameter to control publicNetworkAccess 

### PR DESCRIPTION
**Pull request by jceval**
Adds Optional Parameter to control Public Network Access to Storage Accounts

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: arm/Microsoft.Storage/storageAccounts/deploy.bicep [hunk 1]
```diff
@@ -44,6 +44,13 @@ param storageAccountSku string = 'Standard_GRS'
 @description('Optional. Storage Account Access Tier.')
 param storageAccountAccessTier string = 'Hot'
 
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+@description('Optional. Storage Account Public Network Access.')
+param publicNetworkAccess bool
```
Comment (MrMCake) [inline arm/Microsoft.Storage/storageAccounts/deploy.bicep:52]: If optional, needs a default value

File: arm/Microsoft.Storage/storageAccounts/deploy.bicep [hunk 2]
```diff
@@ -165,7 +172,8 @@ var saBaseProperties = {
   isHnsEnabled: ((!enableHierarchicalNamespace) ? null : enableHierarchicalNamespace)
   minimumTlsVersion: minimumTlsVersion
   networkAcls: (empty(networkAcls) ? null : networkAcls_var)
-  allowBlobPublicAccess: allowBlobPublicAccess
+  allowBlobPublicAccess: (storageAccountKind == 'true') ? null : allowBlobPublicAccess
```
Comment (MrMCake) [inline arm/Microsoft.Storage/storageAccounts/deploy.bicep:175]: Change to
```suggestion
  allowBlobPublicAccess: allowBlobPublicAccess == 'true' ? allowBlobPublicAccess : null
```

File: arm/Microsoft.Storage/storageAccounts/deploy.bicep [hunk 3]
```diff
@@ -165,7 +172,8 @@ var saBaseProperties = {
...
+  publicNetworkAccess: (empty(publicNetworkAccess) ? null : publicNetworkAccess
```
Comment (MrMCake) [inline arm/Microsoft.Storage/storageAccounts/deploy.bicep:176]:
```suggestion
  publicNetworkAccess: !empty(publicNetworkAccess) ? publicNetworkAccess : null
```

File: arm/Microsoft.Storage/storageAccounts/deploy.bicep (hunk 1 above)
Comment (MariusStorhaug) [reply arm/Microsoft.Storage/storageAccounts/deploy.bicep:52]: Should this be Disabled by default?

File: arm/Microsoft.Storage/storageAccounts/deploy.bicep (hunk 2 above)
Comment (MariusStorhaug) [reply arm/Microsoft.Storage/storageAccounts/deploy.bicep:175]: Agree to the swap around, its just to drive consistency on having the change being the "true" approach, while the "false" resulting in no change, being `null`

File: arm/Microsoft.Storage/storageAccounts/deploy.bicep (hunk 3 above)
Comment (MariusStorhaug) [reply arm/Microsoft.Storage/storageAccounts/deploy.bicep:176]: Same as the other comment

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]

File: arm/Microsoft.Storage/storageAccounts/deploy.bicep (hunk 1 above)
Comment (MariusStorhaug) [inline arm/Microsoft.Storage/storageAccounts/deploy.bicep:52]:
```suggestion
param publicNetworkAccess bool = 'Disabled'
```

Comment (MariusStorhaug) [reply arm/Microsoft.Storage/storageAccounts/deploy.bicep:52]: Guess the default in the UI is on

**Comment by MrMCake** [comment]
@MariusStorhaug as nothing came to be so far I'm tempted to close the PR and create a new. Any thoughts?

**Comment by MariusStorhaug** [comment]
Stale PR. Closing and adding changes in another PR.

**Pull request closed without merging**